network/address.go

```diff
@@ -346,20 +346,13 @@ func SelectControllerAddress(addresses []Address, machineLocal bool) (Address, b
 // selection, otherwise just ScopeCloudLocal are.
 func SelectMongoHostPortsBySpaceNames(hostPorts []HostPort, spaces []SpaceName) ([]string, bool) {
```

Comment:
This is only used in one place.
Do we need a utility method when we can call FilterHostPortsBySpaces and do the conversion to slice at the site?

Reply:
fwiw, I don't think I like having  Select *Mongo* ... in 'network/*'. It feels wrong that network would think about the reason for an object, rather than just filtering addresses based on spaces. I'm happy to get rid of it.